extension/src/app.js

```diff
@@ -0,0 +1,178 @@
+import Vue from 'vue';
+import vuetify from './plugins/vuetify' // path to vuetify export
+import App from './App.vue';
+import Vuex from 'vuex'
+// import { sync } from 'vuex-router-sync'
+import VueRouter from "vue-router"
+
+Vue.use(VueRouter)
+Vue.use(Vuex);
+
+function makeid(length) {
+    var result = '';
+    var characters = 'abcdefghijklmnopqrstuvwxyz0123456789';
+    var charactersLength = characters.length;
+    for (var i = 0; i < length; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
+    return result;
+}
+
+import HomeView from "./HomeView.vue";
+import SharingView from "./SharingView.vue";
+
+const routes = [
```

Comment:
We can put the routes in a separate file to make it easier as this file already has too much going on.